packages/labs/analyzer/src/lib/model.ts

```diff
@@ -304,33 +403,64 @@ export type ClassHeritage = {
 export interface ClassDeclarationInit extends DeclarationInit {
   node: ts.ClassDeclaration;
   getHeritage: () => ClassHeritage;
+  fieldMap?: Map<string, ClassField> | undefined;
+  methodMap?: Map<string, ClassMethod> | undefined;
 }
 
 export class ClassDeclaration extends Declaration {
   readonly node: ts.ClassDeclaration;
   private _getHeritage: () => ClassHeritage;
   private _heritage: ClassHeritage | undefined = undefined;
+  readonly _fieldMap: Map<string, ClassField> | undefined;
+  readonly _methodMap: Map<string, ClassMethod> | undefined;
 
   constructor(init: ClassDeclarationInit) {
     super(init);
     this.node = init.node;
     this._getHeritage = init.getHeritage;
+    this._fieldMap = init.fieldMap;
+    this._methodMap = init.methodMap;
   }
 
   get heritage(): ClassHeritage {
     return (this._heritage ??= this._getHeritage());
   }
+
+  get fields() {
+    return Array.from(this._fieldMap?.values() ?? []);
+  }
+
+  get methods() {
+    return Array.from(this._methodMap?.values() ?? []);
+  }
+
+  getField(name: string): ClassField | undefined {
+    return this._fieldMap?.get(name);
+  }
+
+  getMethod(name: string): ClassMethod | undefined {
+    return this._methodMap?.get(name);
+  }
+
+  getMember(name: string): ClassMethod | ClassField | undefined {
```

Comment:
Can you document if these do prototype lookups or just get own properties?

Reply:
Good call, done (for now they only return info on the immediate class, not inherited members). 

I figure we'll add another API or `inherited: boolean` flag later to automatically return the full list, once we have use cases that need it, but didn't want to go crazy adding a bunch of API that we're not using yet.